tests/e2e/scale/test_osd_node_balancing.py

```diff
@@ -0,0 +1,187 @@
+        ):
+            if len(new_osd_list) == new_osd_count:
+                break
+    ceph_health_check(tries=30, delay=60)
+    collect_stats("OSD capacity has been increased")
+
+
+def collect_stats(action_text):
+    """
+    Write the current configuration information into the REPORT file.
+    This information includes the osd, nodes and which osds are on which
+    nodes.  The minimum and maximum numbers of osds per node are also
+    computed and saved.  If this is the final call to collect_stats
+    (action_text parameter is FINAL_REPORT), then the data collected
+    in the REPORT file is also displayed in the log.
+
+    Args:
+        action_text -- Title of last action taken
+                (usually adding nodes or adding osds)
+    """
+    pod_obj = ocp.OCP(
+        kind=constants.POD, namespace=constants.OPENSHIFT_STORAGE_NAMESPACE
+    )
+    osd_list = pod_obj.get(selector=constants.OSD_APP_LABEL)["items"]
+    node_stats = {}
+    for osd_ent in osd_list:
+        osd_node = osd_ent["spec"]["nodeName"]
+        if osd_node in node_stats:
+            node_stats[osd_node].append(osd_ent)
+        else:
+            node_stats[osd_node] = [osd_ent]
+    osds_per_node = []
+    for entry in node_stats:
+        osds_per_node.append(len(node_stats[entry]))
+    wnodes = get_nodes(constants.WORKER_MACHINE)
+    for wnode in wnodes:
+        if wnode.name not in node_stats:
+            osds_per_node.append(0)
+    maxov = max(osds_per_node)
+    minov = min(osds_per_node)
+    this_skew = maxov - minov
+    logging.info(f"Skew found is {this_skew}")
+    logpath = ocsci_log_path()
+    outfile = f"{logpath}/{REPORT}"
+    with open(outfile, "a+") as fd:
+        fd.write(f"\nAction: {action_text}\n\n")
+        fd.write(f"Current OSDS: {len(osd_list)}\n")
+        for entry in osd_list:
+            fd.write(f"\t\t{entry['metadata']['name']}\n")
+        fd.write(f"\nCurrent Nodes: {len(wnodes)}\n")
+        for entry in wnodes:
+            fd.write(f"\t\t{entry.name}\n")
+        fd.write("\nOSD distribution:\n")
+        for entry in osd_list:
+            fd.write(f"\t\t{entry['metadata']['name']}")
+            fd.write(f" is on {entry['spec']['nodeName']}\n")
+        fd.write(f"\nMaximum number of OSDs on a node: {maxov}\n")
+        fd.write(f"Minimum number of OSDs on a node: {minov}\n")
+        fd.write(f"Skew (Difference in OSD counts): {this_skew}\n")
+        fd.write("\n-------------------------------------------------------------\n")
+    balanced = True
+    if this_skew > 1 and maxov > MAX_OSDS_PER_NODE:
+        balanced = False
+    if action_text == FINAL_REPORT or not balanced:
+        with open(outfile, "r") as fd:
+            logged_info = fd.read()
+            for inline in logged_info.split("\n"):
+                logging.info(inline)
+        logging.info(f"Summary of test results output in {outfile}")
+    assert balanced, "OSDs are not balanced"
+
+
+@ignore_leftovers
```

Comment:
If we are expecting the cluster to be destroyed after this test, we should not use left_overs.